### PR TITLE
LibJS+LibTimeZone: Respect the user-provided time zone in Intl.DateTimeFormat

### PR DESCRIPTION
**Pull request by trflynn89**
```js
> Intl.DateTimeFormat([], { timeZoneName: "long", timeZone: "America/New_York" }).format()
"1/12/2022, 8:29 AM Eastern Standard Time"
> Intl.DateTimeFormat([], { timeZoneName: "long", timeZone: "UTC" }).format()
"1/12/2022, 1:29 PM Coordinated Universal Time"
> Intl.DateTimeFormat([], { timeZoneName: "long", timeZone: "Australia/Sydney" }).format()
"1/12/2022, 11:29 PM Australian Eastern Standard Time"
```

### CODE REVIEW

**Comment by trflynn89** [comment]
Making a draft because the default time zone broke some Temporal tests

File: Userland/Libraries/LibJS/Runtime/Temporal/TimeZone.cpp [hunk 1]
```diff
@@ -56,11 +56,11 @@ String canonicalize_time_zone_name(String const& time_zone)
 }
 
 // 11.1.3 DefaultTimeZone ( ), https://tc39.es/proposal-temporal/#sec-defaulttimezone
-// NOTE: This is the minimum implementation of DefaultTimeZone, supporting only the "UTC" time zone.
+// 15.1.3 DefaultTimeZone ( ), https://tc39.es/proposal-temporal/#sec-defaulttimezone
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Temporal/TimeZone.cpp:59]:
```suggestion
// 15.1.3 DefaultTimeZone ( ), https://tc39.es/proposal-temporal/#sup-defaulttimezone
```

Comment (trflynn89) [reply Userland/Libraries/LibJS/Runtime/Temporal/TimeZone.cpp]: Oops, fixed

**Comment by linusg** [comment]
Very cool!

**Pull request merged**